dom/src/main/scala/fs2/dom/ResizeObserver.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2022 Arman Bilge
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs2.dom
+
+import org.scalajs.dom
+import cats.effect.kernel.Async
+import cats.effect.std.Dispatcher
+import cats.effect.kernel.Resource
+import cats.effect.kernel.Sync
+
+abstract class ResizeObserver[F[_]] private {
```

Comment:
Do you need this in Scala 3? If not, then maybe we can just move it into `scala-2/` folder to punt the issue about unsafe `dom.Element` APIs.

Otherwise LGTM, thanks :)

Reply:
No problem at all, I only need it for Scala 2 for the time being.